### ISSUE TITLE
Make interaction with native views more reliable

### ISSUE THREAD

**Issue by bartekpacia** [leancodepl/patrol#409]
**TL;DR** We should introduce a (configurable globally and on a per-case basis) timeout when finding native actions. We should also be loud when the native view is not found (currently we fail silently, which is very bad). 

For example, see #409 as a result of this problem.

### Synopsis

Currently, when the user does:

```dart
await $.native.tap(Selector(text: 'Sign in));
```

the native side checks if a widget with text `'Sign in'` exists, and if it doesn't, we crash. This is not a good DX. 

The scope is not only to improve `tap()` and `enterText()`, but also permission-handling related methods, which are also impacted by this problem.

Goals:
- [ ] add `Duration visibleTimeout` to `NativeAutomator()` constructor. This will be the global timeout that defaults to 10 sec. Send timeouts to native automator during constructor call.
- [ ] wait for `visibleTimeout` instead of crashing immediately when interacting with native views (iOS+Android)

Non-goals (though could be done in the future, when there's a need)
- make `tap()` and `doubleTap()` accept timeout (defaulting to global timeout) on iOS and Android 
- make `enterText()` accept timeout (defaulting to global timeout) on iOS and Android 
- make permission-handling methods accept timeout (global timeout) on iOS and Android

To do in separate issue:
- consider adding timeouts to other functions (e.g open/list/tap notifications, getNativeViews)

See also:
- #248, which might be related

cc: @mateuszwojtczak
